google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java

```diff
@@ -711,7 +714,9 @@ private void appendLoop() {
           // from inflightRequestQueue and prepent them onto the waitinRequestQueue. They need to be
           // prepended as they need to be sent before new requests.
           while (!inflightRequestQueue.isEmpty()) {
-            waitingRequestQueue.addFirst(inflightRequestQueue.pollLast());
+            AppendRequestAndResponse requestWrapper = inflightRequestQueue.pollLast();
+            requestWrapper.requestSendTimeStamp = null;
```

Comment:
I am wondering if you need to set this to null since it seems you will always set it before sending?

Reply:
Whenever a request is removed from the inflight queue, this signals that we no longer expect to receive a response for this request. Hence we should not be checking this request for the timeout limit using a stale send timestamp. There could be a scenario where a request is placed back into the inflight queue, but must wait behind an earlier request that had failed and had a sequence number. We don't want to check it for a timeout limit until it has been sent again.